cmake_modules/FindCuVSLAM.cmake

```diff
@@ -0,0 +1,71 @@
+# - Find cuVSLAM library (https://github.com/NVIDIA-ISAAC-ROS/isaac_ros_visual_slam)
+#
+#  CUVSLAM_ROOT_DIR environment variable can be set to find the library.
+#
+# It sets the following variables:
+#  CUVSLAM_FOUND         - Set to false, or undefined, if cuVSLAM isn't found.
+#  CUVSLAM_INCLUDE_DIRS  - The cuVSLAM include directory.
+#  CUVSLAM_LIBRARIES     - The cuVSLAM library to link against.
+
+# Always do manual search
+if(TRUE)
+    # Fallback to manual search if no CMake config is provided
+    find_path(CUVSLAM_INCLUDE_DIRS 
+        NAMES cuvslam.h
+        PATHS
+            /usr/include
+            /usr/local/include
+            /opt/cuvslam/include
+            /opt/ros/humble/share/isaac_ros_nitros/cuvslam/include
+            $ENV{CUVSLAM_ROOT}/include
+            $ENV{CUVSLAM_ROOT_DIR}/include
+    )
+    
+    find_library(CUVSLAM_LIBRARY 
+        NAMES cuvslam
+        PATHS
+            /usr/lib
+            /usr/local/lib
+            /opt/cuvslam/lib
+            /opt/ros/humble/share/isaac_ros_nitros/cuvslam/lib
+            $ENV{CUVSLAM_ROOT}/lib
+            $ENV{CUVSLAM_ROOT_DIR}/lib
+    )
+    
+    if(CUVSLAM_INCLUDE_DIRS AND CUVSLAM_LIBRARY)
+        set(CUVSLAM_FOUND TRUE)
+        set(CUVSLAM_LIBRARIES ${CUVSLAM_LIBRARY})
+    endif()
+endif()
+
```

Comment:
Following comment above with CUDA found inside that file, you could add the `${CUDA_LIBRARIES}` and `${CUDA_INCLUDE_DIRS}` to `CUVSLAM_LIBRARIES` and `CUVSLAM_INCLUDE_DIRS` variables. So downstream, we can either do:
1. (old cmake way) 
```cmake
IF(CUVSLAM_FOUND)
    SET(INCLUDE_DIRS
		${INCLUDE_DIRS}
		${CUVSLAM_INCLUDE_DIRS}
	)
	SET(LIBRARIES
		${LIBRARIES}
		${CUVSLAM_LIBRARIES}
	)
ENDIF(CUVSLAM_FOUND)
```
2. (modern cmake way, use target) 
```cmake
IF(CUVSLAM_FOUND)
	SET(LIBRARIES
		${LIBRARIES}
		cuvslam::cuvslam
	)
ENDIF(CUVSLAM_FOUND)
```